lib/msal-browser/scripts/README.md

```diff
@@ -0,0 +1,273 @@
+# MSAL Browser Log Decoder
+
+This script helps decode hashed logging strings from the minified MSAL browser bundle back to their original, human-readable messages.
+
+## Background
+
+When MSAL Browser is built in production mode, all logging strings are minified using consistent hashing to reduce bundle size. This means that instead of seeing:
+
+```
+acquireTokenSilent - attempting to acquire token from native platform
+```
+
+Users will see:
+
+```
+ac8ab7
+```
+
+This script allows you to decode these hashes back to the original messages for debugging purposes.
+
+## Features
+
+- **Remote Mapping Fetching**: Automatically fetches log mappings from npm registry for specific versions
+- **Multi-Package Support**: Combines mappings from both `@azure/msal-browser` and `@azure/msal-common` packages
+- **Multiple Input Formats**: Supports various hash input formats (arrays, space-separated, JSON)
+- **Timing Information**: Supports new format with timing data for performance analysis
+- **Verbose Mode**: Optional detailed output for debugging
+- **Caching**: Caches remote mappings locally (24 hours) to reduce network requests
+- **Fallback Support**: Falls back to local mappings when remote fetching fails
+
+## Usage
+
+### Basic Usage
+
+```bash
+# Decode hashes using local mappings
+npm run decode-logs -- [ac8ab7,b2c9d1,e4f6a3]
+
+# Decode with timing information (new format)
+npm run decode-logs -- [0,ac8ab7;5,b2c9d1;10,e4f6a3]
+
+# Space separated hashes
+npm run decode-logs -- ac8ab7 b2c9d1 e4f6a3
+
+# Single hash
+npm run decode-logs -- ac8ab7
+
+# Using stdin
+echo '[ac8ab7,e4f6a3]' | npm run decode-logs
```

Comment:
Suggest adding an example showing how I would pipe a file containing hashes to the script as I expect that will be the most common use case.